client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncDatabase.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2022 Tigris Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tigrisdata.db.client.service;
+
+import com.tigrisdata.db.client.error.TigrisDBException;
+import com.tigrisdata.db.client.model.AlterCollectionResponse;
+import com.tigrisdata.db.client.model.CollectionOptions;
+import com.tigrisdata.db.client.model.CreateCollectionResponse;
+import com.tigrisdata.db.client.model.DropCollectionResponse;
+import com.tigrisdata.db.client.model.TigrisCollectionType;
+import com.tigrisdata.db.client.model.TigrisDBSchema;
+import com.tigrisdata.db.client.model.TransactionOptions;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public interface TigrisAsyncDatabase {
+
+  /**
+   * Return list of collection names
+   *
+   * @return a future to the {@link List} of {@link String} representing collection names
+   * @throws TigrisDBException in case of an error.
+   */
+  CompletableFuture<List<String>> listCollections() throws TigrisDBException;
+
+  /**
+   * Creates a collection under current database.
+   *
+   * @param collectionName name of the collection
+   * @param schema schema of the collection
+   * @param collectionOptions collection option
+   * @return the future to the {@link CreateCollectionResponse}
+   * @throws TigrisDBException in case of an error.
+   */
+  CompletableFuture<CreateCollectionResponse> createCollection(
+      String collectionName, TigrisDBSchema schema, CollectionOptions collectionOptions)
+      throws TigrisDBException;
+
+  /**
+   * Alters a collection under current database.
+   *
+   * @param collectionName name of the collection
+   * @param schema schema of the collection
+   * @param collectionOptions collection option
+   * @return the future to the {@link AlterCollectionResponse}
+   * @throws TigrisDBException in case of an error.
+   */
+  CompletableFuture<AlterCollectionResponse> alterCollection(
```

Comment:
remove it and just have createCollection for now as we will be remove alter from the server side as well.

Reply:
removed AlterCollection API from client.